### ISSUE TITLE
Questions

### ISSUE THREAD

**Issue by kokolll** [RichardSunnyMeng/LatentWatermark-official-codes#5]
I encountered several issues when using the code:

1. **Loading Diffusion Model Weights:**  
   The original code for loading the diffusion model weights is as follows:  
   ```python
   missing, unexpected = self.diffusion.load_state_dict(state_dict=state_dict, strict=False)
   ```
   However, it only works for me if I change it to:  
   ```python
   missing, unexpected = self.diffusion.load_state_dict(state_dict=state_dict['state_dict'], strict=False)
   ```

2. **Watermark Embedding and Extraction:**  
   The embedded watermark is encoded as either `1` or `-1`. However, during extraction, the watermark is processed with the following code, which outputs `1` or `0`:  
   ```python
   if message_model.model.mode == "regression":
       dec_msgs[dec_msgs > 0] = 1
       dec_msgs[dec_msgs <= 0] = 0
   elif message_model.model.mode == "logistic":
       dec_msgs[dec_msgs > 0.5] = 1
       dec_msgs[dec_msgs <= 0.5] = 0
   ```

3. **Training and Evaluation Issues:**  
   I trained the model in three stages using the COCO 5k dataset. During testing, the bit accuracy is around 50%, while the detection accuracy is 1. Upon inspecting the code, I noticed that `thr_eval=0`, which seems to cause any detection accuracy above 0 to be considered successful.  
   Alternatively, this could be due to training issues on my end. Could you consider releasing the pre-trained model weights to help verify this?

**Comment by kokolll** [comment]
I managed to resolve the issue myself by modifying the third-stage training code. Specifically, I replaced this line:

```python
m_out.update(message_model.train_decode(**batch, **g_out, **m_out))
```

To this:
```python
m_out.update(message_model.train_decode(g_out['z_rec']))
```

This change resolved the problem on my end. Please let me know if this modification is consistent with the intended implementation.
Also, I noticed that the detection threshold (thr_eval) is set to 0. Shouldn't this value be different? A threshold of 0 seems unusual in this context.

**Comment by RichardSunnyMeng** [comment]
Thanks for your attention to our work.

For messages, they are originally in binary form and are mapped to {1, -1} for inputs and outputs, i.e. the regression mode. This mapping is to follow a binomial distribution with a mean of 0. The regression mode is the default mode and you can ignore the other.

For the threshold, you can set it according to yourself or follow the recommended value in the paper. And model release is in our to-do list.
